### PR TITLE
feat: Community data

### PR DESCRIPTION
**Pull request by camrun91**
This PR ties the API and the UI together for the Communities feature set.

### CODE REVIEW

**Comment by camrun91** [comment]
The current status of this is the main query is working need to work on the join and leave community queries.

**Comment by camrun91** [comment]
@calebpanza @vinnyjth once this PR is approved https://github.com/ApollosProject/apollos-cluster/pull/290 this one will need to be tested and then will be good.

**Review by calebpanza — APPROVED** [review]
Everything is working on my end! @camrun91 can you do another pass of testing just to confirm it's all looking good?

**Comment by camrun91** [comment]
Pulled this and ran it locally looks good to me!

**Comment by camrun91** [comment]
@vinnyjth just needs your review on this one and community stuff should be good at a baseline.

File: packages/apollos-ui-connected/src/CommunitySingleConnected/components/Details.js [hunk 1]
```diff
@@ -87,6 +88,31 @@ const Details = ({ communityId }) => {
     loading,
   } = useCommunitySingle(communityId);
   const invite = useShare(communityId);
+
+  const handleLeaveCommunity = useMemo(() => {
```
Comment (vinnyjth) [inline packages/apollos-ui-connected/src/CommunitySingleConnected/components/Details.js:92]: you might be better served by `useCallback` :)

**Review by vinnyjth — COMMENTED** [review]
one tiny readability improvement by using `useCallback` beyond that code looks good to me!

**Comment by calebpanza** [comment]
@vinnyjth that's been updated. Good catch!

**Review by vinnyjth — APPROVED** [review]

**Pull request merged**